ompi/mca/btl/tcp/btl_tcp_frag.c

```diff
@@ -99,14 +101,16 @@ OBJ_CLASS_INSTANCE(
 
 size_t mca_btl_tcp_frag_dump(mca_btl_tcp_frag_t* frag, char* msg, char* buf, size_t length)
 {
-    int i, used = 0;
+    int i, used;
 
-    used += snprintf(&buf[used], length - used, "%s frag %p iov_cnt %d iov_idx %d size %lu\n",
-                     msg, (void*)frag, (int)frag->iov_cnt, (int)frag->iov_idx, frag->size);
+    used = snprintf(buf, length, "%s frag %p iov_cnt %d iov_idx %d size %lu\n",
+                    msg, (void*)frag, (int)frag->iov_cnt, (int)frag->iov_idx, frag->size);
+    if (used >= length) return length;
```

Comment:
You may want to follow the pattern used in the `out:` label in `mca_btl_tcp_endpoint_dump` and manually terminate the string to be completely safe.  IIRC, some versions of `snprintf` are buggy and won't reliably `'\0'`-terminate a string when the length parameter is exceeded.